rcl/src/rcl/subscription.c

```diff
@@ -193,6 +209,238 @@ rcl_subscription_get_default_options()
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  const rcl_allocator_t * allocator = &options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+  if (expression_parameters_argc > 100) {
+    RCL_SET_ERROR_MSG("The maximum of expression parameters argument number is 100");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  rmw_ret_t rmw_ret = rmw_subscription_content_filter_options_init(
+    filter_expression,
+    expression_parameters_argc,
+    expression_parameter_argv,
+    allocator,
+    &options->rmw_subscription_content_filter_options
+  );
+
+  return rcl_convert_rmw_ret_to_rcl_ret(rmw_ret);
+}
+
+rcl_ret_t
+rcl_subscription_content_filter_options_set(
+  const char * filter_expression,
+  size_t expression_parameters_argc,
+  const char * expression_parameter_argv[],
+  rcl_subscription_content_filter_options_t * options)
+{
+  if (expression_parameters_argc > 100) {
+    RCL_SET_ERROR_MSG("The maximum of expression parameters argument number is 100");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  const rcl_allocator_t * allocator = &options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t ret = rmw_subscription_content_filter_options_set(
+    filter_expression,
+    expression_parameters_argc,
+    expression_parameter_argv,
+    allocator,
+    &options->rmw_subscription_content_filter_options
+  );
+  return rcl_convert_rmw_ret_to_rcl_ret(ret);
+}
+
+rcl_ret_t
+rcl_subscription_content_filter_options_fini(
+  rcl_subscription_content_filter_options_t * options)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  const rcl_allocator_t * allocator = &options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t ret = rmw_subscription_content_filter_options_fini(
+    &options->rmw_subscription_content_filter_options,
+    allocator
+  );
+
+  return rcl_convert_rmw_ret_to_rcl_ret(ret);
+}
+
+bool
+rcl_subscription_is_cft_enabled(const rcl_subscription_t * subscription)
+{
+  if (!rcl_subscription_is_valid(subscription)) {
+    return false;
+  }
+  return subscription->impl->rmw_handle->is_cft_enabled;
+}
+
+rcl_ret_t
+rcl_subscription_set_content_filter(
+  const rcl_subscription_t * subscription,
+  const rcl_subscription_content_filter_options_t * options
+)
+{
+  RCUTILS_CAN_RETURN_WITH_ERROR_OF(RCL_RET_SUBSCRIPTION_INVALID);
+  RCUTILS_CAN_RETURN_WITH_ERROR_OF(RCL_RET_INVALID_ARGUMENT);
+
+  if (!rcl_subscription_is_valid(subscription)) {
+    return RCL_RET_SUBSCRIPTION_INVALID;
+  }
+
+  rcl_allocator_t * allocator = (rcl_allocator_t *)&subscription->impl->options.allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
```

Comment:
`allocator` is not being used in this function at all, so checking it doesn't seem to make sense

```suggestion
  rcl_allocator_t * allocator = (rcl_allocator_t *)&subscription->impl->options.allocator;
  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
```

Reply:
Yes, this will be removed as mentioned [before](https://github.com/ros2/rcl/pull/894#discussion_r836967104).